adapters/globalsun/globalsun.go

```diff
@@ -0,0 +1,148 @@
+package globalsun
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type reqBodyExt struct {
+	GlobalsunBidderExt reqBodyExtBidder `json:"bidder"`
+}
+
+type reqBodyExtBidder struct {
+	Type        string `json:"type"`
+	PlacementID string `json:"placementId,omitempty"`
+	EndpointID  string `json:"endpointId,omitempty"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var err error
+	adapterRequests := make([]*adapters.RequestData, 0, len(request.Imp))
+
+	for _, imp := range request.Imp {
+		reqCopy := *request
+		reqCopy.Imp = []openrtb2.Imp{imp}
+
+		var bidderExt adapters.ExtImpBidder
+		var globalsunExt openrtb_ext.ImpExtGlobalsun
+
+		if err = json.Unmarshal(reqCopy.Imp[0].Ext, &bidderExt); err != nil {
+			return nil, []error{err}
+		}
+		if err = json.Unmarshal(bidderExt.Bidder, &globalsunExt); err != nil {
+			return nil, []error{err}
+		}
+
+		temp := reqBodyExt{GlobalsunBidderExt: reqBodyExtBidder{}}
+		temp.GlobalsunBidderExt.PlacementID = globalsunExt.PlacementID
+		temp.GlobalsunBidderExt.Type = "publisher"
+
+		finalyImpExt, err := json.Marshal(temp)
+		if err != nil {
+			return nil, []error{err}
+		}
```

Comment:
Optional suggestion - Should we rename finalyImpExt to `extJson` since variable is preparing extension json. Also, I think this could be re-written as - 
```
extJson, err := json.Marshal(reqBodyExt {
    GlobalsunBidderExt: reqBodyExtBidder {
            "PlacementID": globalsunExt.PlacementID,
            "Type": "publisher"
    }
})
if err != nil {
    return err
}
```